src/Microsoft.DotNet.Build.VSTSBuildsApi/VSTSBuildClient.cs

```diff
@@ -0,0 +1,238 @@
+            }
+        }
+
+        /// <summary>
+        /// Update VSTS Build Definition
+        /// </summary>
+        /// <param name="definition"></param>
+        /// <returns>updated definitions id</returns>
+        private async Task<JObject> VstsUpdateDefinitionAsync(JObject definition)
+        {
+            string requestUri = $"DefaultCollection/{definition["project"]["name"].ToString()}/_apis/build/definitions/{definition["id"].ToString()}?api-version=2.0";
+
+            HttpContent content = new StringContent(JsonConvert.SerializeObject(definition), System.Text.Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await GetClient(definition["url"].ToString()).PutAsync(requestUri, content);
+            ProcessResponseStatusCode(response);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+        }
+
+        /// <summary>
+        /// Create VSTS Build Definition
+        /// </summary>
+        /// <param name="jsonObject"></param>
+        private async Task<JObject> VstsCreateDefinitionAsync(JObject definition)
+        {
+            /* Remove definition instance identifiable information */
+            RemoveJObjectToken("revison", definition);
+            RemoveJObjectToken("uri", definition);
+            RemoveJObjectToken("createdDate", definition);
+            RemoveJObjectToken("comment", definition);
+            RemoveJObjectToken("_links", definition);
+            RemoveJObjectToken("authoredBy", definition);
+            RemoveJObjectToken("id", definition);
+
+            string requestUri = $"DefaultCollection/{definition["project"]["name"].ToString()}/_apis/build/definitions?api-version=2.0";
+
+            HttpContent content = new StringContent(JsonConvert.SerializeObject(definition), Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await GetClient(definition["url"].ToString()).PostAsync(requestUri, content);
+            ProcessResponseStatusCode(response);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+
+        }
+
+        /// <summary>
+        /// Retrieve a VSTS Build Definition by name and VSTS folder path
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        private async Task<JObject> VstsRetrieveDefinitionByIdAsync(JObject definition)
+        {
+            string requestUri = $"DefaultCollection/{definition["project"]["name"].ToString()}/_apis/build/definitions/{definition["id"].ToString()}?api-version=2.0";
+
+            HttpResponseMessage response = await GetClient(definition["url"].ToString()).GetAsync(requestUri);
+            ProcessResponseStatusCode(response);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+        }
+
+        /// <summary>
+        /// Retrieve a VSTS Build Definition by name and VSTS folder path
+        /// </summary>
+        /// <param name="name"></param>
```

Comment:
Name is inaccurate--I usually just write a summary unless I have things to say about the params. Other comments have this issue too.